File: insertionSort.js
let myArray = [8, 0, 3, 5, 4.5, 1, 6]

function insertionSort(array) {
    let firstIndex = 1;
    for(let i = firstIndex; i < array.length; i++) {
        insert(array, i);
    }
}

function swap(array, i, j) {
    let temp = array[i];
    array[i] = array[j];
    array[j] = temp;
}

function insert(array, endOfSorted) {
    for(let i = endOfSorted; i > 0; i--) {
        if(array[i] < array[i - 1]) {
            swap(array, i, i - 1);
        } else {
            break;
        }
        console.log(array);
    }
}

insertionSort(myArray)
console.log("final answer:", myArray);

File: insertionSort.py
my_array = [2,7,5,0,4,6]

def insertionSort(array):
    first_index = 1
    for k in range(first_index, len(array)):
        print(k, array[k])
        insert(array, k)

def insert(array, end_of_sorted):
    for k in range(end_of_sorted, 0, -1):
        if array[k] < array[k - 1]:
            array[k], array[k - 1] = array[k - 1], array[k]
        else:
            break

insertionSort(my_array)
print("final answer:", my_array)